### PR TITLE
Initial condition for newton-raphson

### PR DESCRIPTION
**Pull request by ezrayst**
- [x] Affine transformation might give initial condition of newton-raphson outside of [-1, 1], thus creating problems for particles near the edge of an element which might result in failure of finding the natural coordinates of the particle. Thus, initial condition is kept [-1, 1].
- [x] Stress computation from material is checked for NaN (not a number) which usually happens in case of zero divided by zero.

### CODE REVIEW

File: include/particle.tcc [hunk 1]
```diff
@@ -452,8 +452,23 @@ bool mpm::Particle<Tdim, Tnphases>::compute_stress(unsigned phase) {
     if (material_.at(phase) != nullptr) {
       Eigen::Matrix<double, 6, 1> dstrain = this->dstrain_.col(phase);
       // Calculate stress
-      this->stress_.col(phase) = material_.at(phase)->compute_stress(
+      auto stress = material_.at(phase)->compute_stress(
           this->stress_.col(phase), dstrain, this, &state_variables_);
+
```
Comment (kks32) [inline include/particle.tcc]: This is an expensive check to do every time, why is this happening?

File: include/cell.h [hunk 2]
```diff
@@ -8,6 +8,8 @@
 #include <set>
 #include <vector>
 
+#include <math.h>
```
Comment (kks32) [inline include/cell.h]:
```suggestion
#include <cmath>
```

**Review by kks32 — CHANGES_REQUESTED** [review]

File: include/particle.tcc (hunk 1 above)
Comment (ezrayst) [reply include/particle.tcc]: Materials can give NaN stresses. I think I can omit it for this PR, but in MC model, we can have zero divided by zero case, which will give NaN stress, thus NaN acceleration, velocity and location.

**Comment by kks32** [comment]
On new branch `nr`

**Pull request closed without merging**